tests/runtest.py

```diff
@@ -143,20 +169,81 @@
 os.environ["MIDDLEWARE_TEST_PASSWORD"] = passwd
 os.environ["SERVER_TYPE"] = "ENTERPRISE_HA" if ha else "STANDARD"
 
-ip_to_use = ip if not ha else os.environ['controller1_ip']
+ip_to_use = ip
+if ha and ip2:
+    domain = 'tn.ixsystems.com'
+    os.environ['controller1_ip'] = ip
+    os.environ['controller2_ip'] = ip2
 
-with client(host_ip=ip_to_use) as c:
-    interface = c.call(
-        'interface.query',
-        [['state.aliases.*.address', '=', socket.gethostbyname(ip_to_use)]],
-        {'get': True}
-    )['id']
+
+def get_ipinfo(ip_to_use):
+    iface = net = gate = ns1 = ns2 = None
+    with client(host_ip=ip_to_use) as c:
+        net_config = c.call('network.configuration.config')
+        ns1 = net_config.get('nameserver1')
+        ns2 = net_config.get('nameserver2')
+        _ip_to_use = socket.gethostbyname(ip_to_use)
+        for i in c.call('interface.query'):
+            for j in i['state']['aliases']:
+                if j.get('address') == _ip_to_use:
+                    iface = i['id']
+                    net = j['netmask']
+                    for k in c.call('route.system_routes'):
+                        if k.get('network') == '0.0.0.0' and k.get('interface') == i['id']:
```

Comment:
When I'm using `runtest.py` locally against VMs their default gateway is on the NATed interface, but I'm communicating with them (VM-to-VM) on another (host-only) network interface.

Therefore, I added an additional block here:
```
                    for k in c.call('route.system_routes'):
                        if k.get('network') == '0.0.0.0' and k.get('interface') == i['id']:
                            if k['gateway']:
                                return iface, net, k['gateway'], ns1, ns2
                    for k in c.call('route.system_routes'):
                        if k.get('network') == '0.0.0.0':
                            if k['gateway']:
                                return iface, net, k['gateway'], ns1, ns2
```

Acceptable to include, or will I just maintain it out-of-tree for my own purposes?

Reply:
Nope, it's safe to include. I'll push it here, thanks!

Reply:
Fixed 👍